src/components/control/select.jsx

```diff
@@ -147,7 +147,7 @@ const Select = forwardRef(function Select(props, ref) {
       validate();
     }
 
-    if (props.value === value) return;
+    if (JSON.stringify(props.value) === JSON.stringify(value)) return;
```

Comment:
This is pretty difficult to test. If we pass in objects as the option values, if this does regular equality, there will be an infinite loop :-/